alvr/server/cpp/platform/win32/VideoEncoderVCE.cpp

```diff
@@ -180,15 +176,23 @@ amf::AMFComponentPtr VideoEncoderVCE::MakeEncoder(
 		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_USAGE, AMF_VIDEO_ENCODER_USAGE_ULTRA_LOW_LATENCY);
 		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_PROFILE, AMF_VIDEO_ENCODER_PROFILE_HIGH);
 		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_PROFILE_LEVEL, 42);
-		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD, AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD_CBR);
-		// Required for CBR to work correctly
-		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_FILLER_DATA_ENABLE, true);
+		switch (Settings::Instance().m_rateControlMode) {
+			case ALVR_CBR:
+				amfEncoder->SetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD, AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD_CBR);
+				// Required for CBR to work correctly
+				amfEncoder->SetProperty(AMF_VIDEO_ENCODER_FILLER_DATA_ENABLE, true);
+				break;
+			case ALVR_VBR:
+				amfEncoder->SetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD, AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD_LATENCY_CONSTRAINED_VBR);
+				break;
+		}
 		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_TARGET_BITRATE, bitRateIn);
+		amfEncoder->SetProperty(AMF_VIDEO_ENCODER_PEAK_BITRATE, Settings::Instance().m_enableAdaptiveBitrate ? Settings::Instance().m_adaptiveBitrateMaximum : bitRateIn);
```

Comment:
Peak bitrate should probably be equal to target bitrate also for LATENCY_CONSTRAINED_VBR.
See https://raw.githubusercontent.com/GPUOpen-LibrariesAndSDKs/AMF/master/amf/doc/AMF_Video_Encode_HEVC_API.pdf (Table A-3. Default value of parameters).

Reply:
Yes, I mean it should be `bitRateIn` in both cases.

Reply:
Then I guess we also need to adjust peak bitrate in case adaptive bitrate is enabled. Otherwise our target bitrate will be higher than peak at some point. I'll try this change.

Reply:
https://github.com/alvr-org/ALVR/blob/963e09b35742871821e6ff840af7a8c98e2b9794/alvr/server/cpp/platform/win32/VideoEncoderVCE.cpp#L381-L388

This is where the bitrate should be changed when adaptive bitrate is enabled. It's not a property to be set to encoder (at least no encoder currently does it).